packages/core/test/financial-templates/expiring-multiparty/ExpiringMultiPartyCreator.js

```diff
@@ -293,4 +294,27 @@ contract("ExpiringMultiPartyCreator", function(accounts) {
     });
     assert.isTrue(await registry.isContractRegistered(expiringMultiPartyAddress));
   });
+
+  it("Creator can specify a financial product library to transform contract state", async function() {
+    // Create a new FPLib that can transform price and configure the factory to link it with a newly deployed EMP.
+    const structuredNoteFPL = await StructuredNoteFinancialProductLibrary.new();
+    constructorParams.financialProductLibraryAddress = structuredNoteFPL.address;
+
+    // Create the new EMP and grab its saved FPLib.
+    let createdAddressResult = await expiringMultiPartyCreator.createExpiringMultiParty(constructorParams, {
+      from: contractCreator
+    });
+    let expiringMultiPartyAddress;
+    truffleAssert.eventEmitted(createdAddressResult, "CreatedExpiringMultiParty", ev => {
+      expiringMultiPartyAddress = ev.expiringMultiPartyAddress;
+      return ev.expiringMultiPartyAddress != 0 && ev.deployerAddress == contractCreator;
+    });
+    let expiringMultiParty = await ExpiringMultiParty.at(expiringMultiPartyAddress);
+    let linkedFPLib = await StructuredNoteFinancialProductLibrary.at(
+      await expiringMultiParty.financialProductLibrary()
+    );
+
+    // FPLib address is saved correctly.
+    assert.equal(linkedFPLib.address, structuredNoteFPL.address);
+  });
```

Comment:
@UMAprotocol/eng Should any other tests be done here to test that the EMP is "correct linked" to the FPLib?

Reply:
Can try calling a transformation on a deployed emp that uses the transformation such as one of the public view functions? Could also check that the deployed emp has the library set correctly to the address you created

Reply:
You've done the second test actually already

Reply:
IMO, the unit tests for the emp should cover whether the address returned by the public financialProductLibrary() method match the transformations done in the contract, and that sort of test shouldn't be necessary for the scope of this contract. The creator tests should just be sure that the address is injected properly.

But more tests wouldn't necessarily hurt :).

Reply:
In response to first suggestion see [here](https://github.com/UMAprotocol/protocol/pull/2256#issuecomment-737513983)!